lib/archethic/mining.ex

```diff
@@ -96,6 +100,47 @@ defmodule Archethic.Mining do
     validation_node_public_keys == Enum.map(validation_nodes, & &1.last_public_key)
   end
 
+  @doc """
+  Request storage node to lock the mining of this transaction address and hash
+  """
+  @spec request_chain_lock(tx :: Transaction.t()) :: :ok | {:error, :already_locked}
+  def request_chain_lock(tx = %Transaction{address: address, type: type}) do
+    storage_nodes = Election.storage_nodes(address, P2P.authorized_and_available_nodes())
+    nb_storage_nodes = length(storage_nodes)
+
+    hash =
+      tx
+      |> Transaction.to_pending()
+      |> Transaction.serialize()
+      |> Crypto.hash()
+
+    message = %RequestChainLock{address: address, hash: hash}
+
+    aggregated_responses =
+      Task.Supervisor.async_stream_nolink(
+        Archethic.TaskSupervisor,
+        storage_nodes,
```

Comment:
I wondering if we should use locally available nodes, to save some times in the StartMining, as authorized and globally available nodes might not respond due to unavailability

Reply:
Done